usr/src/cmd/zdb/zdb.c

```diff
@@ -3164,8 +3164,8 @@ zdb_dump_block(char *label, void *buf, uint64_t size, int flags)
 	for (i = 0; i < nwords; i += 2) {
 		(void) printf("%06llx:  %016llx  %016llx  ",
 		    (u_longlong_t)(i * sizeof (uint64_t)),
-		    (u_longlong_t)(do_bswap ? BSWAP_64(d[i]) : d[i]),
-		    (u_longlong_t)(do_bswap ? BSWAP_64(d[i + 1]) : d[i + 1]));
+		    (u_longlong_t)BE_64(d[i]),
+		    (u_longlong_t)BE_64(d[i + 1]));
```

Comment:
I think the original intent of this code was to display the values *as 64-bit numbers*.  This is useful if the data is in fact 64-bit numbers (e.g. indirect blocks).  The current code does this correctly, although the header is confusing/incorrect.  `do_bswap` makes sense in this context - you would set it if the data on disk is from the opposite endianness.

I think you're trying to make it so that in the absence of `do_bswap`, the header shows the *byte* numbers, and it shows the low bytes first (byte 0,1,2,...).  I'm not sure I understand the intent / purpose of `do_bswap` in this context.